projects/opyn/gamma.js

```diff
@@ -31,87 +35,173 @@ module.exports = async function tvl(timestamp, block) {
   let balances = {};
 
     if(block >= START_BLOCK) {
-      try{
-      const supportedTokens = await (
-        sdk
-          .api
-          .util
-          .tokenList()
-          .then((supportedTokens) => supportedTokens.map(({ contract }) => contract))
-      );
-
-      // get ETH balance
-      const balance = (await sdk.api.eth.getBalance({target: marginPool, block})).output;
-      balances[ETH] = BigNumber(balances[ETH] || 0).plus(BigNumber(balance)).toFixed();
-
-      await Promise.all(supportedTokens.map(async (pulseToken) => {
-        const collateralAsset = pulseToken.toLocaleLowerCase();
-
-        if(collateralAsset.substring(0, 2) == "0x" && collateralAsset != ETH) {
-          let isWhitelistedCollateralToken = (
+      const whitelistLogs = (
+        await sdk.api.util
+          .getLogs({
+            keys: [],
+            toBlock: block,
+            target: whitelist,
+            fromBlock: START_BLOCK,
+            topics: [whitelistTopic],
+          })
+      ).output;
+
+      let collaterals = [];
+      for (let log of whitelistLogs) {
+        let collateral = `0x${log.topics[1].substr(26)}`.toLowerCase();
+        if (collaterals.indexOf(collateral) < 0) {
+          collaterals.push(collateral);
+        }
+      }
```

Comment:
What we want probably is a Set
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set
Bonus point it saves 6 lines of code :smile: 
```js
const collaterals = new Set(whitelistLogs.map((log) => `0x${log.topics[1].substr(26)}`.toLowerCase())
```
collaterals is now a set, but if you really prefer to be an array at the end, you can spread `[...collaterals]`

Reply:
If we're already shaving the yak... Using `Array.from(collaterals)` should avoid a copy while converting the `Set` to array. :)
Also `array.includes()` should be available in our version of node.